Code/Mantid/Framework/API/inc/MantidAPI/AnalysisDataService.h

```diff
@@ -133,10 +133,17 @@ class DLLExport AnalysisDataServiceImpl
   template <typename WSTYPE>
   boost::shared_ptr<WSTYPE> retrieveWS(const std::string &name) const {
     // Get as a bare workspace
-    boost::shared_ptr<Mantid::API::Workspace> workspace =
-        Kernel::DataService<API::Workspace>::retrieve(name);
-    // Cast to the desired type and return that.
-    return boost::dynamic_pointer_cast<WSTYPE>(workspace);
+    try {
+      boost::shared_ptr<Mantid::API::Workspace> workspace =
+          Kernel::DataService<API::Workspace>::retrieve(name);
+      // Cast to the desired type and return that.
+      return boost::dynamic_pointer_cast<WSTYPE>(workspace);
+    } catch (...) {
```

Comment:
Maybe you should catch only `Kernel::Exception::NotFoundError`? This new message is for the most common exception that the workspace is not found, but there could be other exceptions (memory issues, etc.) and for those you don't want to show this message.

Also, don't forget your pull request labels ;)